dask/dataframe/groupby.py

```diff
@@ -531,7 +531,9 @@ def _mul_cols(df, cols):
     # Fix index in a groupby().apply() context
     # https://github.com/dask/dask/issues/8137
     # https://github.com/pandas-dev/pandas/issues/43568
-    _df.index = [0] * len(_df)
+    # Make sure index dtype is "int64" (even if _df is empty)
+    # https://github.com/dask/dask/pull/9701
+    _df.index = np.repeat(np.array([0], dtype="int64"), len(_df))
```

Comment:
Minor comment -- this seems a bit easier to reason about 

```suggestion
    _df.index = np.zeros(len(_df), dtype=int)
```

Reply:
This one is going to keep me up at night :) I’m not sure I could have found a more roundabout version of np.zeros